### DISCUSSION TITLE
Blame pages (especially failed to load blame pages) should offer links to normal(blob) view

### DISCUSSION THREAD

**Discussion by jsoref** [community/community#41722] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Bug

### Body

Here's a perfectly reasonable request to show a file at a specific version with a specific line:
https://github.com/dart-lang/sdk/blame/24e4e16ef1ec4bf38dd8d01e4af852ee70a837d5/runtime/vm/object.cc#L765

<img width="1227" alt="image" src="https://user-images.githubusercontent.com/2119212/207581338-eeb5ef6d-a8bd-4ebd-be1f-c900e0ba678a.png">

The reason that the request uses /blame/ is that if it used the default view and the file was special (markdown, svg, _insert ever changing list of random files for which github has special handling_ (mermaid?)) then the linked line information would be discarded.

From this page (Blame), one can reach:
1. History
2. Raw
3. Blame (?!)
4. github.dev
5. github desktop
6. a copy of the incredibly long file you can't see (?!)

What you can't reach?
* Blob: https://github.com/dart-lang/sdk/blob/24e4e16ef1ec4bf38dd8d01e4af852ee70a837d5/runtime/vm/object.cc#L765
* The path for the file in case you wanted to easily load it somewhere else or perhaps jump to it using `f` -- the copy button does the wrong thing, and it isn't particularly easy to get the path out of breadcrumbs (whitespace + repository) or out of the url (you have to find the right splice point and lop off the anchor and any params in case there were params as well)

**Comment by Akash1134** [comment]
Hi @jsoref , Thanks for reporting this issue. Currently, this is the intended behavior as blame can sometimes take a very long time depending on the code since it has to compile code across multiple branches and changes. I have shared this feedback with the concerned engineers so that they can look into it.

**Reply by jsoref** [reply]
To be clear, I don't mind GitHub defending itself against large painful files. My notes are about the rest of the UX elements.

**Comment by Marcono1234** [comment]
> The reason that the request uses /blame/ is that if it used the default view and the file was special (markdown, svg, insert ever changing list of random files for which github has special handling (mermaid?)) then the linked line information would be discarded.

At least for Markdown files they added the [`?plain=1` URL parameter](https://github.blog/changelog/2021-06-30-parameter-to-disable-markdown-rendering/) to disable Markdown rendering, e.g.: https://github.com/dart-lang/sdk/blob/analyzer-0.33.1/README.md?plain=1#L15

**Reply by jsoref** [reply]
@Marcono1234 that doesn't work on GitHub Mobile and there are lots of reasons I'm not using it here.

I'm well aware of it.

**Reply by Marcono1234** [reply]
Ok, I just wanted to point this out here since it was not mentioned yet. Though this was not intended as complete solution to your issue with the blame page; I agree with you that it would be useful to handle problems with loading the blame page in a better way.

> and there are lots of reasons I'm not using it here

Out of curiosity, do you mind sharing some of them?

**Reply by jsoref** [reply]
I'm in a hurry and don't have time today. I might be able to dig them up next week.

But given that your link did the wrong thing in GitHub Mobile, and it also increases the length of my urls (I do hit size limits, so every character counts), none of these are good reasons to use it.
